### ISSUE TITLE
Local files with special characters (spaces, accents, etc) fail to open in iOS

### ISSUE THREAD

**Issue by awfulcode** [Bilibili/ijkplayer#553]
Hi guys,

I'm trying to find out where things break down, but there's a bug when trying to open local files that have special characters. This is what shows up in the log:

ijkmp_prepare_async()=0
ijkmp_start()
ijkmp_start()=-3
file:///var/mobile/Containers/Data/Application/4B988DD9-047E-4737-A698-775FE77ED0FA/Documents/testing%20more%20spaces.MOV: No such file or directory
2015-10-14 17:30:44.402 IJKMediaDemo[4900:1459564] FFP_MSG_ERROR: 0
2015-10-14 17:30:44.402 IJKMediaDemo[4900:1459564] moviePlayBackStateDidChange 2: paused
2015-10-14 17:30:44.402 IJKMediaDemo[4900:1459564] playbackStateDidChange: MPMovieFinishReasonPlaybackError: 1

I'll keep trying to find where this is going wrong, but it be wonderful if you could help. Thanks!

**Comment by awfulcode** [comment]
Well, at least I figured out what's throwing the error. This line in ffplay.c:

    err = avformat_open_input(&ic, is->filename, is->iformat, &ffp->format_opts);

Still working on how to fix it. I'm targeting iOS 8+ and it seems that it has changed the documents folder, so maybe that has something to do with it. We'll see...

**Comment by bbcallen** [comment]
Local path seems being escaped as URL
Try "initWithContentURLString" instead of "initWithContentURL"

**Comment by awfulcode** [comment]
Good morning! I figured it out last night. This is the patch if you want to implement it:

```
- (id)initWithContentURL:(NSURL *)aUrl
             withOptions:(IJKFFOptions *)options
     withSegmentResolver:(id<IJKMediaSegmentResolver>)segmentResolver
{
    if (aUrl == nil)
        return nil;
    
    // Detect if URL is file path and return proper string for it
    NSString *initUrl = [aUrl isFileURL] ? [aUrl path] : [aUrl absoluteString];
    
    return [self initWithContentURLString:initUrl
                              withOptions:options
                      withSegmentResolver:segmentResolver];
}
```

**Comment by bbcallen** [comment]
Merged: c539907bf65b3477aafdfc0bd69865361c2cea03
Thanks.
